SpringBootCrudApi/src/main/java/com/cengenes/configuration/api/controller/OrderController.java

```diff
@@ -0,0 +1,162 @@
+            @RequestParam(value = "customerid", required = false) Integer customerId,
+            @RequestParam(value = "employeeid", required = false) Integer employeeId,
+            @RequestParam(value = "status", required = false) String orderStatus,
+            @RequestHeader(value = "Authorization") String authorization, HttpServletResponse response,
+            Pageable pageable
+    ) {
+        authService.authenticate(authorization, response);
+        OrderInfoResponse resp = new OrderInfoResponse();
+        if (response.getStatus() == 200) {
+            OrderInfo qry = new OrderInfo();
+            if (orderId != null) {
+                qry.setOrderId(orderId);
+            }
+            if (customerId != null) {
+                qry.setCustomerId(customerId);
+            }
+            if (employeeId != null) {
+                qry.setEmployeeId(employeeId);
+            }
+            if (orderStatus != null) {
+                qry.setOrderStatus(orderStatus);
+            }
+
+            Page<OrderInfo> pg = orderInfoRepository.findAll(org.springframework.data.domain.Example.of(qry), pageable);
+            resp.setPageStats(pg, true);
+            resp.setItems(pg.getContent());
+        } else {
+            resp.setOperationStatus(OperationResponse.ResponseStatusEnum.NO_ACCESS);
+            resp.setOperationMessage("No access");
+        }
+
+        return resp;
+    }
+
+
+    @GetMapping(value = "/order-details")
+    public OrderDetailResponse getOrderDetail(@RequestParam(value = "orderid", required = false) Integer orderId, @RequestHeader(value = "Authorization") String authorization, HttpServletResponse response) {
+        authService.authenticate(authorization, response);
+        OrderDetailResponse resp = new OrderDetailResponse();
+        if(response.getStatus() == 200){
+            long prevOrderId = -1, newOrderId;
+            OrderDetail orderDetail = new OrderDetail();
+            int itemCount = 0;
+
+
+            resp.setItems(new ArrayList<OrderDetail>());
+            String sql = " select order_id, book_id   , customer_id   , order_date, order_status  , shipped_date    , employee_id , payment_type, paid_date, "
+                    + " ship_name      , ship_address1, ship_address2 , ship_city , ship_state    , ship_postal_code, ship_country, "
+                    + " book_code   , book_name , category      , quantity  , unit_price    , discount        , date_allocated, order_item_status, "
+                    + " shipping_fee   , customer_name, customer_email, customer_company from A_TEAM.ORDER_DETAILS ";
+            String where = " where 1 = 1 ";
+            String order = " order by order_id, book_id ";
+            if (orderId != null) {
+                where = where + " and order_id = " + orderId;
+            }
+
+            List<Map<String, Object>> list = jdbcTemplate.queryForList(sql + where + order);

```

Comment:
Opppss! Where is the clean code :) You can move all business inside service layer.Also please use Java 8 futeres like lambda,streams.https://www.baeldung.com/java-8-streams